tests/js/server/dump/dump-cluster.js

```diff
@@ -280,6 +280,70 @@ function dumpTestSuite () {
         assertEqual({ value: [ i, i ] }, doc.more);
       }
     },
+////////////////////////////////////////////////////////////////////////////////
+/// @brief test padded keygen
+////////////////////////////////////////////////////////////////////////////////
+
+    testKeygenPadded : function () {
+      var c = db._collection("UnitTestsDumpKeygenPadded");
+      var p = c.properties();
+
+      assertEqual(2, c.type()); // document
+      assertFalse(p.waitForSync);
+      assertEqual("padded", p.keyOptions.type);
+      assertFalse(p.keyOptions.allowUserKeys);
+      assertEqual(process.env['paddedLastValue'], p.keyOptions.lastValue);
+
+      assertEqual(1, c.getIndexes().length); // just primary index
+      assertEqual("primary", c.getIndexes()[0].type);
+      assertEqual(1001, c.count());
+
+      let allDocs = {};
+      c.toArray().forEach(doc => {
+        allDocs[doc.value] = doc;
+      });
+      let lastKey = "";
+      for (var i = 0; i < 1000; ++i) {
```

Comment:
Shouldn't this be 1001? We're checking here 1000 docs in total but count in Line 299 says 1001

Reply:
the 1001 document doesn't adhere the similar syntax. Its the 1002'nd document, and here to revalidate that 1001 was successfully deleted.